hazelcast/src/main/java/com/hazelcast/sql/SqlQuery.java

```diff
@@ -0,0 +1,286 @@
+/*
+ * Copyright (c) 2008-2020, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.sql;
+
+import com.hazelcast.config.SqlConfig;
+import com.hazelcast.internal.util.Preconditions;
+
+import javax.annotation.Nonnull;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * Definition of a SQL query.
+ * <p>
+ * This object is mutable. Properties are read once before the execution is started.
+ * Changes to properties do not affect the behavior of already running queries.
+ */
+public class SqlQuery {
+    /** Value for the timeout that is not set. */
+    public static final long TIMEOUT_NOT_SET = -1;
+
+    /** Default timeout. */
+    public static final long DEFAULT_TIMEOUT = TIMEOUT_NOT_SET;
+
+    /** Default page size. */
+    public static final int DEFAULT_CURSOR_BUFFER_SIZE = 4096;
+
+    private String sql;
+    private List<Object> parameters;
+    private long timeout = DEFAULT_TIMEOUT;
+    private int cursorBufferSize = DEFAULT_CURSOR_BUFFER_SIZE;
+
+    public SqlQuery(String sql) {
```

Comment:
We should also mark this argument as nonnull. Also elsewhere.

Reply:
Fixed.